test/fixtures/es-module-loaders/hooks-input.mjs

```diff
@@ -17,7 +17,11 @@ export async function resolve(specifier, context, next) {
   if (resolveCalls === 1) {
     url = new URL(specifier).href;
     assert.match(specifier, /json-modules\.mjs$/);
-    assert.strictEqual(context.parentURL, undefined);
+
+    if (context.parentURL && !context.parentURL.includes('eval')) {
```

Comment:
`parentURL.includes('eval')` is too broad a check; it could capture a filename like `test-eval.mjs` or something. Can you make it an exact check for whatever `parentURL` is set to for the `--eval` case? Something like `[eval]` or whatever it is?

Reply:
(I think there's also a generation id in there or something similar)

Reply:
It could be a regex like `/^\[eval\d*\]$/.test(context.parentURL)` or whatever matches the actual string.

Reply:
Heads up: when triggered from `--eval`, `parentURL` != `` `[eval${evalIndex}]` ``; it's that but appended to the path to the current node executable, so I removed the starting `^`. We _could_ try to do something fancier like including a separator but that gets complicated real quick (thanks Windows), so I think it's sufficient to assume nobody is naming a file `[eval1]`.

Reply:
This is just a test fixture, so I think that’s fine.